Gems/EMotionFX/Code/EMotionFX/Source/MotionSet.cpp

```diff
@@ -734,38 +721,25 @@ namespace EMotionFX
     }
 
 
-    uint32 MotionSet::GetNumChildSets() const
+    size_t MotionSet::GetNumChildSets() const
     {
         MCore::LockGuardRecursive lock(m_mutex);
 
-        uint32 childSetSize = 0;
-        for (const MotionSet* motionSet : m_childSets)
+        return AZStd::accumulate(begin(m_childSets), end(m_childSets), size_t{0}, [](size_t total, const MotionSet* motionSet)
         {
-            if (!motionSet->GetIsOwnedByRuntime())
-            {
-                ++childSetSize;
-            }
-        }
-        return childSetSize;
+            return total + motionSet->GetIsOwnedByRuntime();
+        });
     }
 
 
-    MotionSet* MotionSet::GetChildSet(uint32 index) const
+    MotionSet* MotionSet::GetChildSet(size_t index) const
     {
         MCore::LockGuardRecursive lock(m_mutex);
-        uint32 currentIndex = 0;
-        for (MotionSet* motionSet : m_childSets)
+        const auto foundChildSet = AZStd::find_if(begin(m_childSets), end(m_childSets), [iter = index](const MotionSet* motionSet) mutable
         {
-            if (!motionSet->GetIsOwnedByRuntime())
-            {
-                if (currentIndex == index)
-                {
-                    return motionSet;
-                }
-                ++currentIndex;
-            }
-        }
-        return nullptr;
+            return !motionSet->GetIsOwnedByRuntime() && iter-- == 0;
+        });
```

Comment:
I used std algorithms to describe what the code is doing rather than how it does it. This code is finding an element in a vector, so `find_if` makes sense. It has the added logic of not finding the first match, but the `index`th match.